var/spack/repos/builtin/packages/python/package.py

```diff
@@ -892,7 +895,7 @@ def headers(self):
         except ProcessError:
             headers = find_headers(
                 'pyconfig', self.prefix.include, recursive=True)
-            config_h = headers[0]
+            config_h = headers[0] if len(headers) > 0 else ''
```

Comment:
Have you encountered times when this didn't work?

Reply:
I am a colleague of Dan Street and I can chime in with some more details. This error was first noticed when debugging with the `--fake` flag with `spack install`. A simple way to produce the unexpected IndexError is to run:

```
spack install --fake python
spack load python
python
==> Error: list index out of range
```

This is outside the scope of the PR and I can remove it if need be

Reply:
Hmm, I guess I would expect `spack install --fake <package> && spack load <package>` to fail regardless of which package you choose. Is there some reason you would like to enable that functionality? What are you using it for?

Reply:
You're right this is a silly example. I talked to the member of my team that originally made this change and he is unsure why he added it in. I will remove the change. My mistake